sycl/doc/EnvironmentVariables.md

```diff
@@ -25,6 +25,9 @@ compiler and runtime.
 | `SYCL_EAGER_INIT` | Integer | Enable by specifying non-zero value. Tells the SYCL runtime to do as much as possible initialization at objects construction as opposed to doing lazy initialization on the fly. This may mean doing some redundant work at warmup but ensures fastest possible execution on the following hot and reportable paths. It also instructs PI plugins to do the same. Default is "0".  |
 | `SYCL_REDUCTION_PREFERRED_WORKGROUP_SIZE` | See [below](#sycl_reduction_preferred_workgroup_size) | Controls the preferred work-group size of reductions. |
 | `SYCL_ENABLE_FUSION_CACHING` | '1' or '0' | Enable ('1') or disable ('0') caching of JIT compilations for kernel fusion. Caching avoids repeatedly running the JIT compilation pipeline if the same sequence of kernels is fused multiple times. Default value is '1'. |
+| `SYCL_JIT_AMDGCN_PTX_KERNELS` | '1' or '0' | Enable ('1') or disable ('0') JIT compilation of kernels. Only supported for Nvidia and AMD backends. Note, that it is required to have a valid binary for the desired backend (AMD or CUDA), that was compiled with `-fsycl-embed-ir` in order to use JIT-ing. When JIT-ing is enabled SYCL runtime will try to cache and reuse JIT-compiled kernels, furthermore if a kernel uses specialization constants the compiler will attempt to materialize the values in place, turning them to de-facto compile time constants. Default is '0'. |
```

Comment:
Could we imagine a world where a user would want jitting for only AMDGCN or PTX and have the other be regular AOT compilation?

Reply:
It does sound possible, I am not sure how likely, but definitely possible. I am not aware of any AMD and Nvidia systems that we currently support (but that's not quite the same as to say that there are none). The binary format that we compile for is [taken from the device](https://github.com/jchlanda/llvm/blob/jakub/jit_spec_const/sycl/source/detail/jit_compiler.cpp#L95), so that's independent of the options. As it stands now, the JITting is "sticky", once set it will affect the behaviour until it is unset.

I can see the following possibilities here:

1. leave it as it is,
2. explode the flags to only address one backend at a time,
3. tap in to clang options parsing/handling and try to reuse that logic.

Exploding the flags doesn't seem to be a great design, so I'd guess it would have to be either option 1, or 3 (with a cavity that 3 should probably be a follow-up).

Reply:
I see. I'm fine with keeping it as-is then.